### ISSUE TITLE
Error message while creating subscription lost due to socket.close() maximum payload size

### ISSUE THREAD

**Issue by Jonahss** [enisdenjo/graphql-ws#73]
**Screenshot**
```
(node:48769) UnhandledPromiseRejectionWarning: RangeError: The message must not be greater than 123 bytes
    at Sender.close (/Users/jonahss/Workspace/cloud-platform/api/node_modules/graphql-ws/node_modules/ws/lib/sender.js:120:15)
    at WebSocket.close (/Users/jonahss/Workspace/cloud-platform/api/node_modules/graphql-ws/node_modules/ws/lib/websocket.js:233:18)
    at WebSocket.onMessage (/Users/jonahss/Workspace/cloud-platform/api/node_modules/graphql-ws/cjs/server.js:326:28)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
```

**Expected Behaviour**
If an error is thrown, I expect some message about what I did wrong

**Actual Behaviour**
In this case, graphql-ws tries to send my client the error message from the server, but there is a limit of 123 bytes for the message of a socket close payload. So my client gets this RangeError instead.

**Debug Information**
This error gets thrown right here: https://github.com/enisdenjo/graphql-ws/blob/ec1b461641c769b2678da1b4876673c2395a1b3e/src/server.ts#L710

**Further Information**
I guess just truncate the error message? Should probably log the full error plus stacktrace on the server side?

I ran into this error because I returned a rather large object inside my `onConnect()` handler. This object was not JSON serializable because it contained a cycle. The error warning about the cycle was longer than 123 bytes. Should I create a separate issue for how to handle passing in non-serializable payloads?

**Comment by enisdenjo** [comment]
Great catch! Thanks for reporting this.

I aim to increase the server control with #64. In essence, once the PR lands, you'll have full power over making error handling decisions. By providing your own server implementation - its up to you to decide if you want to close the socket and report the error, ignore it, perform a mitigation or simply alter the result.
